common/frame.go

```diff
@@ -770,6 +770,47 @@ func (f *Frame) uncheck(selector string, opts *FrameUncheckOptions) error {
 	return err
 }
 
+// IsChecked returns true if the first element that matches the selector
+// is checked. Otherwise, returns false.
+func (f *Frame) IsChecked(selector string, opts goja.Value) bool {
+	f.log.Debugf("Frame:IsChecked", "fid:%s furl:%q sel:%q", f.ID(), f.URL(), selector)
+
+	popts := NewFrameIsCheckedOptions(f.defaultTimeout())
+	if err := popts.Parse(f.ctx, opts); err != nil {
+		k6ext.Panic(f.ctx, "%w", err)
+	}
+	checked, err := f.isChecked(selector, popts)
+	if err != nil {
+		k6ext.Panic(f.ctx, "%w", err)
+	}
+
+	return checked
+}
+
+func (f *Frame) isChecked(selector string, opts *FrameIsCheckedOptions) (bool, error) {
+	isChecked := func(apiCtx context.Context, handle *ElementHandle) (interface{}, error) {
+		v, err := handle.isChecked(apiCtx, 0) // Zero timeout when checking state
+		if errors.Is(err, ErrTimedOut) {      // We don't care about timeout errors here!
+			return v, nil
+		}
+		return v, err
+	}
+	act := f.newAction(
+		selector, DOMElementStateAttached, opts.Strict, isChecked, []string{}, false, true, opts.Timeout,
+	)
+	v, err := callApiWithTimeout(f.ctx, act, opts.Timeout)
```

Comment:
If the timeout is not wanted when checking state, should we still pass `opts.Timeout` to `newAction` and `callApiWithTimeout`? Does it make sense to call `callApiWithTimeout` at all?

This seems like an awful lot of code for what can essentially be done in a single line with `Page.evaluate()`... I wonder if we need all of this. :thinking:

Reply:
🤔 Hmm, good point. We need to make a refactoring for all the related methods in the `Frame` as well if we are going to follow this route. I created a new issue for this https://github.com/grafana/xk6-browser/issues/374.